lib/sycamore/sycamore/query/planner.py

```diff
@@ -129,6 +130,105 @@ def process_json_plan(
     return result_nodes[0], nodes
 
 
+def postprocess_llm_helper(user_message: str, llm_client: LLM) -> str:
+    messages = [
+        {
+            "role": "system",
+            "content": """You are a helpful agent that assists in small transformations
+                    of input text as per the instructions. You should make minimal changes
+                    to the provided input and keep your response short""",
+        },
+        {"role": "user", "content": user_message},
+    ]
+
+    prompt_kwargs = {"messages": messages}
+    chat_completion = llm_client.generate(prompt_kwargs=prompt_kwargs, llm_kwargs={"temperature": 0, "seed": 42})
+    return chat_completion
+
+
+def postprocess_plan(llm_plan: Any, llm_client: LLM) -> Any:
+    """Given the query plan provided by the LLM, postprocess it using a set of rules
+    that modify the plan for optimizing or other purposes."""
+
+    if isinstance(llm_plan, str):
+        postprocessed_plan = extract_json(llm_plan)
+    else:
+        postprocessed_plan = copy.deepcopy(llm_plan)
+
+    # Rule 1: If the plan has a vector search in the beginning followed by a count or nothing or extract_entity,
+    # we replace that by removing the vector search and adding an LLM Filter before the count
+
+    if postprocessed_plan[0]["operatorName"] == "QueryVectorDatabase" and (
+        len(postprocessed_plan) == 1 or postprocessed_plan[1]["operatorName"] in ["Count", "LlmExtractEntity"]
+    ):
+        # If the first operator has an "opensearch_filter", we will convert it to a QueryDatabase with
+        #        that opensearch_filter as "query"
+        # If not, we do a QueryDatabase with "match_all" instead
+        op = postprocessed_plan[0]
+
+        modified_description = postprocess_llm_helper(
```

Comment:
I'm very surprised you need an LLM call for this. Can you explain?

Reply:
it's just changing the description which can say this is a vector search, not the actual node object

Reply:
As Vinayak said, we need to convert the original description like "find all incidents that were caused by environmental conditions" to "find all incidents" -- assuming we want the description to be correct, an LLM would have to be used to manipulate the text properly..